src/Framework/EventFile.cxx

```diff
@@ -7,18 +7,15 @@
 #include "Framework/EventFile.h"
 #include "Framework/Exception/Exception.h"
 #include "Framework/RunHeader.h"
-#include "Recon/Event/EventConstants.h"
 
 namespace framework {
 
-EventFile::EventFile(const std::string &filename, EventFile *parent,
-                     bool isOutputFile, bool isSingleOutput,
-                     int compressionSetting, bool isLoopable)
-    : fileName_(filename),
-      parent_(parent),
-      isOutputFile_(isOutputFile),
-      isSingleOutput_(isSingleOutput),
-      isLoopable_(isLoopable) {
+EventFile::EventFile(const framework::config::Parameters &params,
```

Comment:
While this works and I don't have a good reason for removing this, I don't like passing parameters to EventFile. To me, it is confusing to have the EventFile get some of its parameters from the python class for Process. The solution to this confusion is either to have an EventFile python class for configuring files or to not provide EventFile with a Parameters instance.

Reply:
Again, I agree with you in principle, but the changes required to clean this up are outside the scope of this PR. Let's discuss and plan on these changes for a v4.0 release.